src/audio/module_adapter/module_adapter.c

```diff
@@ -35,7 +35,21 @@ struct comp_dev *module_adapter_new(const struct comp_driver *drv,
 	int ret;
 	struct comp_dev *dev;
 	struct processing_module *mod;
-	struct ipc_config_process *ipc_module_adapter = spec;
+	unsigned char *data;
+	uint32_t size;
+
+	switch (config->type) {
+	case SOF_COMP_MODULE_ADAPTER:
+	{
+		struct ipc_config_process *ipc_module_adapter = spec;
+
+		size = ipc_module_adapter->size;
+		data = ipc_module_adapter->data;
+		break;
+	}
+	default:
+		return NULL;
```

Comment:
do we expect to add more types here? I was in favour of using `switch` for all enumerations even if only one case has to be handled, but we do use `if` for such cases too and I must admit that looks simpler, shorter and just as readable, so maybe here
```
if (config->type != SOF_COMP_MODULE_ADAPTER)
	return NULL;
```
would be easier too, but that's totally up to you

Reply:
I believe there _is_ indeed such an expectation so I'd say this is fine.

Reply:
yes @lyakh the adapter will need to handle other types. The switch/case will be expanded as and when I convert the current components